code/modules/mod/mod.dm

```diff
@@ -0,0 +1,555 @@
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+			return FALSE
+		install(attacking_item, FALSE)
+		return TRUE
+	else if(istype(attacking_item, /obj/item/stock_parts/cell))
+		if(!open)
+			audible_message("<span class='warning'>[src] indicates that it needs to be open before installing [attacking_item].</span>", "<span class='warning'>[src] flashes an error message.</span>")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+			return FALSE
+		if(cell)
+			audible_message("<span class='warning'>[src] indicates that there is a cell already installed.</span>", "<span class='warning'>[src] flashes an error message.</span>")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+			return FALSE
+		attacking_item.forceMove(src)
+		cell = attacking_item
+		audible_message("<span class='notice'>[src] indicates that [cell] has been succesfully installed.</span>", "<span class='warning'>[src] flashes a confirmation message.</span>")
+		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+		return TRUE
+	else if(is_wire_tool(attacking_item) && open)
+		wires.interact(user)
+		return TRUE
+	else if(istype(attacking_item, /obj/item/mod/paint) && paint(user, attacking_item))
+		to_chat(user, "<span class='notice'>You paint [src] with [attacking_item].</span>")
+		qdel(attacking_item)
+		return TRUE
+	else if(open && attacking_item.GetID())
+		update_access(attacking_item)
+		return TRUE
+	return ..()
+
+/obj/item/mod/control/get_cell()
+	if(!wearer)
+		return cell
+
+/obj/item/mod/control/emag_act(mob/user)
+	locked = !locked
+	to_chat(user, "<span class='notice'>You emag [src], [locked ? "locking" : "unlocking"] it.</span>")
+
+/obj/item/mod/control/doStrip(mob/stripper, mob/owner)
+	toggle_activate(stripper, TRUE)
+	for(var/part in modules)
+		conceal(stripper, part)
+	return ..()
+
+/obj/item/mod/control/proc/paint(mob/user, obj/item/paint)
+	if(theme.skins.len <= 1)
+		return FALSE
+	var/list/display_names = list()
+	var/list/skins = list()
+	for(var/mod_skin in theme.skins)
+		display_names[mod_skin] = REF(mod_skin)
+		var/image/skin_image = image(icon = icon, icon_state = "[mod_skin]-control")
+		skins += list(mod_skin = skin_image)
+	var/pick = show_radial_menu(user, src, skins, custom_check = FALSE, require_near = TRUE)
+	if(!pick || !user.is_holding(paint))
+		return FALSE
+	var/skin_reference = display_names[pick]
+	var/new_skin = locate(skin_reference) in theme.skins
+	skin = new_skin
+	var/list/skin_updating = mod_parts.Copy() + src
+	for(var/obj/item/piece as anything in skin_updating)
+		piece.icon_state = "[skin]-[initial(piece.icon_state)]"
+	wearer.update_icons()
+	return TRUE
+
+/obj/item/mod/control/proc/shock(mob/living/user)
+	if(!istype(user) || cell.charge < 1)
+		return FALSE
+	do_sparks(5, TRUE, src)
+	var/check_range = TRUE
+	return electrocute_mob(user, get_area(src), src, 0.7, check_range)
```

Comment:
```suggestion
	return electrocute_mob(user, get_area(src), src, 0.7, TRUE)
```

Reply:
uh, shouldn't this be using the suit's cell, not the power of whatever area the suit is in?

Reply:
good catch thank you